internal/js_parser/js_parser_lower.go

```diff
@@ -268,35 +268,26 @@ func (p *parser) lowerFunction(
 					*args = append(*args, js_ast.Arg{Binding: js_ast.Binding{Loc: arg.Binding.Loc, Data: &js_ast.BIdentifier{Ref: argRef}}})
 				}
 
-				// Forward all arguments from the outer function to the inner function
-				if p.fnOnlyDataVisit.argumentsRef != nil {
```

Comment:
The problem seems to be `argumentsRef` is non-nil if the async arrow function is nested inside a regular function. This code path only triggers when the parameters have a default value (and hence `couldThrowErrors = true`), so both conditions are required for this bug.

Reply:
Thanks for the summary.